inspirehep/records/api/base.py

```diff
@@ -246,9 +246,9 @@ def get_subclasses(cls):
     @classmethod
     def get_record(cls, id_, with_deleted=False):
         record = super().get_record(str(id_), with_deleted)
-        type_from_schema = PidStoreBase.get_pid_type_from_schema(record["$schema"])
-        if record.pid_type is None or record.pid_type != type_from_schema:
-            return cls.get_subclasses()[type_from_schema].get_record(id_, with_deleted)
+        record_class = cls.get_class_for_record(record)
+        if record_class != cls:
+            return record_class.get_record(id_, with_deleted)
```

Comment:
Am I missing something, but can we just ``record_class(record)``? do we need a second query here? ☺️

Reply:
https://github.com/inveniosoftware/invenio-records/blob/master/invenio_records/api.py#L33